### PR TITLE
test: allowance unit tests

### PR DESCRIPTION
**Pull request by zzmp**

### CODE REVIEW

File: src/hooks/usePermitAllowance.test.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+const EXPIRATION = 1234567890
+const NONCE = 42
+
+jest.mock('hooks/multicall')
+jest.mock('hooks/useContract')
+const mockUseContract = useContract as jest.Mock
+const mockUseSingleCallResult = useSingleCallResult as jest.Mock
+
+describe('usePermitAllowance', () => {
+  const CONTRACT = new Contract(PERMIT2_ADDRESS, PERMIT2_ABI) as Permit2
```
Comment (vm) [inline src/hooks/usePermitAllowance.test.ts:29]: nit: can these live top level?

File: src/hooks/usePermitAllowance.test.ts [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+      expect(result.current).toMatchObject({ permitAllowance: undefined })
+    })
+  })
+
+  describe('with no allowance', () => {
+    beforeEach(() =>
+      mockUseSingleCallResult.mockReturnValue({ result: { amount: BigInt(0), expiration: EXPIRATION, nonce: NONCE } })
+    )
+
+    it('refetches allowance every block', () => {
```
Comment (vm) [inline src/hooks/usePermitAllowance.test.ts:56]: i'm not following what's going on w this one

File: src/hooks/usePermitAllowance.test.ts (hunk 1 above)
Comment (zzmp) [reply src/hooks/usePermitAllowance.test.ts]: done. also moved some other (related) const's to the top level.

File: src/hooks/usePermitAllowance.test.ts (hunk 2 above)
Comment (zzmp) [reply src/hooks/usePermitAllowance.test.ts]: added an explanation to the test name.

**Review by vm — APPROVED** [review]

**Pull request merged**